File: shopping_android_app/app/src/main/java/com/example/shopping_android_app/adapter/CategoryListAdapter.java
package com.example.shopping_android_app.adapter;

import android.content.Context;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.shopping_android_app.R;
import com.example.shopping_android_app.base.BaseAdapter;
import com.example.shopping_android_app.model.home.HomeBean;

import java.util.List;

public class CategoryListAdapter extends BaseAdapter {

    private final Context context;

    public CategoryListAdapter(Context context, List<HomeBean.DataBean.CategoryListBean.GoodsListBean> data) {
        super(context, data);
        this.context = context;
    }

    @Override
    protected int getLayout(int type) {
        return R.layout.home_item_category;
    }

    @Override
    protected void bindData(Object data, VH vh) {
        ImageView image = (ImageView) vh.getViewById(R.id.new_pic_url);
        TextView name = (TextView) vh.getViewById(R.id.name);
        TextView floor_price = (TextView) vh.getViewById(R.id.floor_price);

        HomeBean.DataBean.CategoryListBean.GoodsListBean bean= (HomeBean.DataBean.CategoryListBean.GoodsListBean) data;

        Glide.with(context).load(bean.getList_pic_url()).into(image);
        name.setText(bean.getName());
        floor_price.setText("￥"+bean.getRetail_price());
    }
}

File: kotlin/app/src/main/java/com/example/myapplication/adapter/home/CategoryListAdapter.kt
package com.example.myapplication.adapter.home

import android.content.Context
import android.util.SparseArray
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.databinding.library.baseAdapters.BR
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.myapplication.R
import com.example.myapplication.base.BaseAdapter
import com.example.myapplication.base.IItemClick
import com.example.myapplication.data.Category
import com.example.myapplication.data.Goods
import com.example.myapplication.home.HomeFragment
import com.shop.base.BaseActivity
import kotlinx.android.synthetic.main.home_item_topic.view.*

class CategoryListAdapter(
    context: Context?,
    list: List<Goods>,
    layouts : SparseArray<Int>,
    itemClick: IItemClick<Goods>
) : BaseAdapter<Goods>(context, list,layouts,itemClick) {
//    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
//        return BaseViewHolder(DataBindingUtil.inflate(LayoutInflater.from(context),
//            viewType,
//            parent,
//            false))
//    }
//
//    override fun onBindViewHolder(holder: BaseViewHolder, position: Int) {
//        val get = list.get(position)
//        var bindingUtil: ViewDataBinding = holder.dataBinding
//        bindingUtil.setVariable(BR.vmGoods, get)
//        with(holder!!.itemView) {
//            Glide.with(context).load(get.list_pic_url)?.into(new_pic_url)
//        }
//    }
//
//    override fun getItemCount(): Int = list.size
//
//    override fun getItemViewType(position: Int): Int {
//        return R.layout.home_item_categorygoods
//    }

    /**
     * 加载完数据刷新到界面的data
     */
    fun refreshData(it: List<Goods>) {
        list = it
        notifyDataSetChanged()
    }

    override fun layoutId(position: Int): Int {
        return R.layout.home_item_categorygoods
    }

    override fun bindData(binding: ViewDataBinding, data: Goods, layoutId: Int) {
        binding.setVariable(BR.vmGodosClick,itemClick)
    }
}